### ISSUE TITLE
TangemSdk.sign invokes both Success and Failure

### ISSUE THREAD

**Issue by muratogat** [tangem/tangem-sdk-android-old#336]
Using:
```
implementation "com.github.tangem.tangem-sdk-android:core:3.8.2"
implementation "com.github.tangem.tangem-sdk-android:android:3.8.2"
```

When I try to get something signed using TangemSdk.sign , after successfully signing and getting the CompletionResult.Success , also the CompletionResult.Failure is invoked unexpectedly.

To be clear, the the Success part works OK and has the correct signature. So things seem to work fine, until the Failure is also triggered. It doesn't cause problems when the failure is ignored, but if a failure message is displayed or some code is run when Tangem signing fails, it breaks the proper operation.

The code I receive in the Failure TangemSdkError is 50002 . 
Perhaps this has something to do with cards setup with the older Tangem app. Perhaps not.

Could you please check?

**Comment by KostasZav** [comment]
Hey Tangem team, any updates here?

**Comment by KostasZav** [comment]
@lazutkin-andrey @mobile-tangem can you please provide some feedback about this issue?

**Comment by kozarezvlad** [comment]
We'll check it and give feedback.

**Comment by KostasZav** [comment]
@lazutkin-andrey @mobile-tangem @kozarezvlad any updates here?

**Comment by kozarezvlad** [comment]
Check it in version 3.9.1.

Code 50002 happens when dialog has been closed by user interaction with BottomSheet, otherwise it shouldn't throws.

If error still happens in 3.9.1, let us know.

**Comment by muratogat** [comment]
Yes unfortunately I still see this behavior in 3.9.1
I am definitely not closing the popup with user interaction. The flow is:

- App requests signature
- I trigger TangemSDK 
- Card scan popup is displayed, I scan the card
- Pin is asked, I enter the pin
- The success sheet is shown with the check, then it closes itself
- To my wallet both Success and Failure callbacks are returned, with the Success having the correct signature and Failure having the code 50002

What could be the reason that 50002 is triggered even though the user doesn't close this popup manually?

**Comment by kozarezvlad** [comment]
We're looking your case

**Comment by kozarezvlad** [comment]
Fix has been merged in develop branch. 
You can try it with tag `develop-350`

**Comment by muratogat** [comment]
Yes, I confirm this solves the issue.
Many thanks. Please kindly merge into next main release.

**Comment by muratogat** [comment]
When could we expect a release including this fix? 
I would really not like to ship a version that has a develop version of TangemSDK integrated. Thanks.

**Comment by kozarezvlad** [comment]
Fixed in 3.9.2
